### PR TITLE
Rework git tutorial and slides into unified marp presentation

### PR DESCRIPTION
**Pull request by ripleymj**
Here's the draft for tomorrow night. I'd like to merge this before the meeting so it's published on our site, but can modify it before or do a second commit after.

### CODE REVIEW

File: git101/index.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+---
+# What is git?
+
+- Git stores a series of changes to files and attributes of that change
+  - File diff, author, timestamp, previous change hash
+- Change and attributes are hashed together to uniquely identify a commit
+- Branches allow different authors to pursue different paths
+  - Like an actual tree, branches can go their own way, but the trunk must be shared
+- Never delete a commit, use a revert to undo, but that change is then logged
```
Comment (lam2mo) [inline git101/index.md:93]:
```suggestion
- Never delete a commit: use a revert to undo, but that change is then logged
```

File: git101/index.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+---
+# Creating a repository on the GitHub website
+
+Using the + button in the top right corner or the webpage, choose
+"New Repository". Pick a name and optionally a description. For this tutorial,
+we will not be using a pre-provided README, .gitignore, or license.
+
+Select the HTTPS button towards the top of this guide, and notice the section
+for `push an existing repository`. Don't need to do anything with this now.
```
Comment (lam2mo) [inline git101/index.md:127]:
```suggestion
for `push an existing repository`. No need to do anything with this now.
```

File: git101/index.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+Just keep the page open.
+
+---
+# Starting your first `git` repository
+
+First, we will create a directory to use for this tutorial.
+
+    mkdir -p ~/UUG/uug-git-intro && cd $_
+
+If we try to use a git command here, we will get an error
```
Comment (lam2mo) [inline git101/index.md:137]:
```suggestion
If we try to use a git command here, we will get an error:
```

File: git101/index.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+    git commit -am "Test commit"
+
+Notice that you got an error when you tried to do this! You cannot create a
+commit in git without that directory being a part of a git repository.
+
+---
+# Starting your first `git` repository
+
+So let's initialize your git repository. This starts getting `git` set up to
+run.
```
Comment (lam2mo) [inline git101/index.md:148]:
```suggestion
Let's initialize your git repository.
```

File: git101/index.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+---
+# Demo Time
+
+---
+# Creating a repository on the GitHub website
+
+Using the + button in the top right corner or the webpage, choose
+"New Repository". Pick a name and optionally a description. For this tutorial,
+we will not be using a pre-provided README, .gitignore, or license.
```
Comment (lam2mo) [inline git101/index.md:124]:
```suggestion
we will not be using a pre-provided README, .gitignore, or license.
Remember the name; you will need this later!
```

File: git101/index.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+    git init
+
+If you run `ls -al`, you will now see a `.git` directory there. This is where
+git stores all of the data about your repository.
+
+---
+# Starting your first `git` repository
+
+You now have to tell `git` where your code should be pushed to. We are using
+GitHub for this and we'll now we will use that command from earlier.
```
Comment (lam2mo) [inline git101/index.md:159]:
```suggestion
GitHub for this. Run the following command, but change `YOURUSERNAME` to your
GitHub user name and change `REPONAME` to the repository name you chose earlier.
```

File: git101/index.md [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+    git remote add origin https://github.com/YOURUSERNAME/REPONAME.git
+
+Remember later that you have named your GitHub URL `origin`.
+
+---
+# Basic `git` configuration
+
+By default, `git` will have your name and email configured to be your login
+name@hostname. This is nice in some ways, but probably not your actual
+email address. Let's fix this.
```
Comment (lam2mo) [inline git101/index.md:170]:
```suggestion
email address. Let's fix this. Make sure you substitute all of your own info
in the following commands:
```

File: git101/index.md [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+
+---
+# Making your first commit
+
+Commits are basically a record of changes to your directory. Each time you
+make a change, you need to tell `git` to create a new record. You do this
+with the commit command.
+
+Before you commit, you have to make a change to the directory. Let's create
+a file
```
Comment (lam2mo) [inline git101/index.md:187]:
```suggestion
a file:
```

File: git101/index.md [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+
+    vim README.txt
```
Comment (lam2mo) [inline git101/index.md:189]:
```suggestion
    nano README.txt
```

Let's avoid having to extricate newbies from vim via Teams! :)

File: git101/index.md [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+    git commit
+
+Vim (or your default editor) will open. Type a description of your changes,
+called a commit message, and save the file (in Vim, you can type `:wq` in
+normal mode).
+
+---
+# See a record of commits
+
+`git` keeps a record of each of your changes in a log. TO see this, let's run
```
Comment (lam2mo) [inline git101/index.md:224]:
```suggestion
`git` keeps a record of each of your changes in a log. To see this:
```

File: git101/index.md [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+# Synchronizing GitHub
+
+The first time you push to an upstream copy, you can use the `-u` option to tell
+git your preferred default.
+
+    git push -u origin main
+
+This will tell git to push your current branch `main` (since you just renamed it)
+to the `main` branch at the `origin` URL, and to remember that for future use. Now
+that you've set the default, you can simple use
```
Comment (lam2mo) [inline git101/index.md:246]:
```suggestion
that you've set the default, you can simply use:
```

File: git101/index.md [hunk 12]
```diff
@@ -0,0 +1,299 @@
...
+    git rm README.txt
+    git commit
+
+---
+Later, we realize that we regret this change. We can undo this. The way that
+reverting works in `git` is by taking the state of the repository at a point in
+time and bringing it back to the present.
+
+So what we need to do is check our log, and find the hash of the commit we want
+to revert
```
Comment (lam2mo) [inline git101/index.md:269]:
```suggestion
to revert. Use the following command (replacing the hash):
```

File: git101/index.md [hunk 13]
```diff
@@ -0,0 +1,299 @@
...
+
+    git log
+    git revert "hash"
+
+This will create a commit with the default message, 'Revert "Old Commit"'. Push
+this change back to GitHub with
```
Comment (lam2mo) [inline git101/index.md:275]:
```suggestion
this change back to GitHub:
```

File: git101/index.md [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+
+    git push
+
+---
+# gitignore
```
Comment (lam2mo) [inline git101/index.md:280]:
```suggestion
# gitignore

```

File: git101/index.md [hunk 15]
```diff
@@ -0,0 +1,299 @@
...
+- While git can contain binary files (like the images in this presentation),
+tradition says it should not contain binaries that are artifacts of source files
+  - Commit your C or Java files, not executables, .o, or .class files
```
Comment (lam2mo) [inline git101/index.md:283]:
```suggestion
  - Commit your C or Java files, not executables, `.o`, or `.class` files
```

**Review by lam2mo — COMMENTED** [review]
Nice--thanks for reworking this!

File: git101/index.md [hunk 16]
```diff
@@ -0,0 +1,301 @@
...
+
+---
+# A note on branching and language
+
+- Git-based projects have a primary branch, traditionally known as `master`
+- You can develop new code on different branches, with the goal of integrating
+that code back into the primary branch
+- Terminology and language matter, and git's use of the term `master` is
+considered insensitive and has been deprecated
+- Consider naming your primary branch `main` or `develop` on new projects
```
Comment (kylelaker) [inline git101/index.md:104]: Is Git 2.28 in Ubuntu 20.04? https://github.blog/2020-07-27-highlights-from-git-2-28/#introducing-init-defaultbranch

File: git101/index.md [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+
+---
+# Help along the way
+
+- Git uses man pages extensively to document its options and features
+- One man page would be too much for all of git, so pages are hyphenated with
```
Comment (kylelaker) [inline git101/index.md:110]: Each command also has a `--help` flag which usually opens up that particular commands `man` page.

File: git101/index.md (hunk 16 above)
Comment (ripleymj) [reply git101/index.md:104]: 2.25 https://packages.ubuntu.com/focal/git

File: git101/index.md (hunk 17 above)
Comment (ripleymj) [reply git101/index.md:110]: No GNU options allowed here

**Pull request merged**